src/rgw/rgw_rest_s3.cc

```diff
@@ -4393,30 +4385,122 @@ RGWAuthApplier::aplptr_t RGWLDAPAuthEngine::authenticate() const
   RGWUserInfo user_info;
   user_info.user_id = base64_token.id;
   if (rgw_get_user_info_by_uid(store, user_info.user_id, user_info) >= 0) {
-    if (user_info.type != TYPE_LDAP) {
+    /*if (user_info.type != TYPE_LDAP) {
       ldout(cct, 10) << "ERROR: User id of type: " << user_info.type << " is already present" << dendl;
       return nullptr;
-    }
+    }*/
   }
```

Comment:
I will enable the type checking once we have a plan to migrate existing users to the new format.